### PR TITLE
feat(server): REST API and WS API now use env vars for host and port.

### PR DESCRIPTION
**Pull request by twinsant**
User description
Background

Previous versions used hard-coded settings for the server's host and port, which was inconvenient during development
Changes 🏗️

Added host and port settings to the .env file

### CODE REVIEW

File: rnd/autogpt_server/.env.example [hunk 1]
```diff
@@ -12,6 +12,15 @@ REDIS_PASSWORD=password
...
 ENABLE_CREDIT=false
 APP_ENV="local"
+
+# REST API
+AP_SERVER_HOST="0.0.0.0"
+AP_SERVER_PORT="8000"
+
+# WS API
+WS_SERVER_HOST="0.0.0.0"
+WS_SERVER_PORT="8001"
```
Comment (kcze) [inline rnd/autogpt_server/.env.example]: nit for consistency
```suggestion
AP_SERVER_HOST=localhost
AP_SERVER_PORT=8000

# WS API
WS_SERVER_HOST=localhost
WS_SERVER_PORT=8001
```
edit: I realised that `0.0.0.0` is not the same as `localhost` but quote marks can be removed nonetheless.

**Review by kcze — APPROVED** [review]

**Comment by kcze** [comment]
I think it would be good if @aarushik93 and @majdyz looked at this as well.

**Comment by Pwuts** [comment]
Since #8008, the ports are configurable:
https://github.com/Significant-Gravitas/AutoGPT/blob/f8f6bbc9a2f5c321c2946e0410e9994ad5ece803/rnd/autogpt_server/autogpt_server/util/settings.py#L75-L88

These settings can be used like `AGENT_SERVER_PORT=8123`.

Do you really need the host to be separately configurable?

**Comment by twinsant** [comment]
> Since #8008, the ports are configurable:
> 
> https://github.com/Significant-Gravitas/AutoGPT/blob/f8f6bbc9a2f5c321c2946e0410e9994ad5ece803/rnd/autogpt_server/autogpt_server/util/settings.py#L75-L88
> 
> These settings can be used like `AGENT_SERVER_PORT=8123`.
> 
> Do you really need the host to be separately configurable?

Yes, on production server, we need listen to the specific internal host for security reasons, so it's better to config host as a setting.

**Comment by twinsant** [comment]
Waiting for a loooooong time....

**Comment by majdyz** [comment]
@twinsant That config file is already env-var configurable, you can use the caps too.
e.g:
https://github.com/Significant-Gravitas/AutoGPT/blob/198a1048e8b8f365180415dfc0fb02818111be87/autogpt_platform/backend/backend/util/settings.py#L103C5-L103C19

If you want to change the Agent API port from 8006 to 8009, you can set:
AGENT_API_PORT=8009

**Comment by majdyz** [comment]
Subsequently, you can utilize the same convention to introduce host configuration.
The current PR is unfortunately clashing with the recent changes, would you mind rebasing your change to adapt to the latest code?

**Comment by majdyz** [comment]
@twinsant [[feat(platform): Make REST & WS server host configurable by majdyz · Pull Request #8143 · Significant-Gravitas/AutoGPT](https://github.com/Significant-Gravitas/AutoGPT/pull/8143/files)](https://github.com/Significant-Gravitas/AutoGPT/pull/8143/files) If you don't mind adapting your code to this, I can close this PR.

**Comment by twinsant** [comment]
> @twinsant #8143 If you don't mind adapting your code to this and see if it matches your requirement, I can close my PR and use yours.

fine, will improve on your changes.

**Pull request closed without merging**